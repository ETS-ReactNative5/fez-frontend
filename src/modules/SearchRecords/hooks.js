import { useHistory, useLocation } from 'react-router';
import deparam from 'can-deparam';
import param from 'can-param';
import { pathConfig } from 'config';
import { locale } from '../../locale';
import { PUB_SEARCH_BULK_EXPORT_SIZE } from '../../config/general';

/**
 * Parse provided query string and return active filters, facets etc
 * @returns object
 */
export const parseSearchQueryStringFromUrl = (queryString, canBulkExport = false, isUnpublishedBufferPage = false) => {
    if (!queryString.length) {
        return {};
    }

    const providedSearchQuery = deparam(queryString);
    if (providedSearchQuery.hasOwnProperty('activeFacets')) {
        providedSearchQuery.activeFacets = {
            filters: providedSearchQuery.activeFacets.filters || {},
            ranges: providedSearchQuery.activeFacets.ranges || {},
            ...(providedSearchQuery.activeFacets.hasOwnProperty('showOpenAccessOnly')
                ? { showOpenAccessOnly: providedSearchQuery.activeFacets.showOpenAccessOnly === 'true' }
                : {}),
        };
    } else {
        providedSearchQuery.activeFacets = {
            filters: {},
            ranges: {},
        };
    }

    const pageSize = parseInt(providedSearchQuery.pageSize, 10);
    if (canBulkExport && pageSize === PUB_SEARCH_BULK_EXPORT_SIZE) {
        providedSearchQuery.bulkExportSelected = true;
        providedSearchQuery.pageSize = PUB_SEARCH_BULK_EXPORT_SIZE;
    } else {
        providedSearchQuery.bulkExportSelected = false;
        providedSearchQuery.pageSize = locale.components.sorting.recordsPerPage.indexOf(pageSize) < 0 ? 20 : pageSize;
    }

    providedSearchQuery.sortDirection =
        locale.components.sorting.sortDirection.indexOf(providedSearchQuery.sortDirection) < 0
            ? locale.components.sorting.sortDirection[0]
            : providedSearchQuery.sortDirection;

    providedSearchQuery.sortBy =
        locale.components.sorting.sortBy.map(sortBy => sortBy.value).indexOf(providedSearchQuery.sortBy) < 0
            ? locale.components.sorting.sortBy[1].value
            : providedSearchQuery.sortBy;

    if (!isUnpublishedBufferPage && !!providedSearchQuery.searchQueryParams) {
        delete providedSearchQuery.searchQueryParams.rek_status;
        delete providedSearchQuery.searchQueryParams.rek_created_date;
        delete providedSearchQuery.searchQueryParams.rek_updated_date;
    }

    providedSearchQuery.page = (providedSearchQuery.page && parseInt(providedSearchQuery.page, 10)) || 1;
    return providedSearchQuery;
};

/**
 * @param searchFields
 * @return {string[]}
 */
export const getAdvancedSearchFields = searchFields => {
    const excludesFromLocale = locale.pages.searchRecords.facetsFilter.excludeFacetsList;
    // Iterate the searchfields and add their map from locale into the excluded facets array
    const importedFacetExcludes = [];
    searchFields.map(searchFieldItem => {
        if (searchFieldItem.searchField) {
            const fieldType = locale.components.searchComponent.advancedSearch.fieldTypes[searchFieldItem.searchField];
            if (fieldType.map) {
                importedFacetExcludes.push(fieldType.map);
            }
        }
    });
    return excludesFromLocale.concat(importedFacetExcludes);
};

/**
 * @param showOpenAccessOnly
 * @return Object
 */
export const queryParamsDefaults = showOpenAccessOnly => ({
    activeFacets: {
        filters: {},
        ranges: {},
        showOpenAccessOnly: showOpenAccessOnly,
    },
    advancedSearchFields: [],
    bulkExportSelected: false,
    page: 1,
    pageSize: 20,
    sortBy: locale.components.sorting.sortBy[1].value,
    sortDirection: locale.components.sorting.sortDirection[0],
});

/**
 *
 * @param showOpenAccessOnly
 * @param queryString
 * @param canBulkExport
 * @param isUnpublishedBufferPage
 * @return Object
 */
export const getQueryParams = (showOpenAccessOnly, queryString, canBulkExport, isUnpublishedBufferPage) => ({
    ...queryParamsDefaults(showOpenAccessOnly),
    ...parseSearchQueryStringFromUrl(queryString, canBulkExport, isUnpublishedBufferPage),
});

/**
 * @param activeFacets
 * @param canBulkExport
 * @param isUnpublishedBufferPage
 * @return Object
 */
export const useQueryStringParams = (showOpenAccessOnly, canBulkExport, isUnpublishedBufferPage) => {
    const history = useHistory();
    const location = useLocation();

    const queryParams = getQueryParams(
        showOpenAccessOnly,
        location.search.substr(1),
        canBulkExport,
        isUnpublishedBufferPage,
    );

    const updateQueryString = queryParams => {
        console.log('updateQueryString:' + param(queryParams));
        history.push({
            pathname:
                location.pathname === pathConfig.admin.unpublished
                    ? pathConfig.admin.unpublished
                    : pathConfig.records.search,
            search: param(queryParams),
        });
    };

    return {
        queryParams: queryParams,
        updateQueryString,
    };
};

/**
 * @param searchQueryParams
 * @param updateQueryParamsState
 * @param actions
 * @return Object
 */
export const useSearchRecordsControls = (searchQueryParams, updateQueryParamsState, actions) => {
    const pageSizeChanged = pageSize => {
        updateQueryParamsState({
            ...searchQueryParams,
            pageSize,
            page: 1,
        });
    };

    const pageChanged = page => {
        updateQueryParamsState({
            ...searchQueryParams,
            page,
        });
    };

    const sortByChanged = (sortBy, sortDirection) => {
        updateQueryParamsState({
            ...searchQueryParams,
            sortBy,
            sortDirection,
        });
    };

    const facetsChanged = activeFacets => {
        updateQueryParamsState({
            ...searchQueryParams,
            activeFacets,
            page: 1,
        });
    };

    const handleExport = exportFormat =>
        actions.exportEspacePublications({
            ...searchQueryParams,
            ...exportFormat,
        });

    const handleFacetExcludesFromSearchFields = searchFields => {
        !!searchFields &&
            updateQueryParamsState({
                ...searchQueryParams,
                advancedSearchFields: getAdvancedSearchFields(searchFields),
            });
    };

    return {
        pageSizeChanged,
        pageChanged,
        sortByChanged,
        facetsChanged,
        handleExport,
        handleFacetExcludesFromSearchFields,
    };
};
