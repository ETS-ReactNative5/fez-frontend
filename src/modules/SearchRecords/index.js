export { default as SearchRecords } from './containers/SearchRecords';
