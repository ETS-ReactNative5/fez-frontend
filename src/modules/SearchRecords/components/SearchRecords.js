import React from 'react';
import PropTypes from 'prop-types';
import param from 'can-param';
import deparam from 'can-deparam';

import Grid from '@material-ui/core/Grid';
import Hidden from '@material-ui/core/Hidden';

import { StandardCard } from 'modules/SharedComponents/Toolbox/StandardCard';
import { StandardPage } from 'modules/SharedComponents/Toolbox/StandardPage';
import { StandardRighthandCard } from 'modules/SharedComponents/Toolbox/StandardRighthandCard';
import { SearchComponent } from 'modules/SharedComponents/SearchComponent';
import { InlineLoader } from 'modules/SharedComponents/Toolbox/Loaders';
import { Alert } from 'modules/SharedComponents/Toolbox/Alert';
import {
    PublicationsList,
    PublicationsListPaging,
    PublicationsListSorting,
    FacetsFilter,
} from 'modules/SharedComponents/PublicationsList';
import { BulkExport } from 'modules/BulkExport';

import { locale } from 'locale';
import { RecordsSelectorContext } from 'context';
import { userIsAdmin, userIsResearcher } from 'hooks';
import { PUB_SEARCH_BULK_EXPORT_SIZE } from 'config/general';
import { pathConfig } from 'config';

/**
 * Parse provided query string and return active filters, facets etc
 * @returns object
 */
const parseSearchQueryStringFromUrl = (searchQuery, canBulkExport, isUnpublishedBufferPage) => {
    const providedSearchQuery = deparam(searchQuery);

    if (providedSearchQuery.hasOwnProperty('activeFacets')) {
        providedSearchQuery.activeFacets = {
            filters: providedSearchQuery.activeFacets.filters || {},
            ranges: providedSearchQuery.activeFacets.ranges || /* istanbul ignore next */ {},
            ...(providedSearchQuery.activeFacets.hasOwnProperty('showOpenAccessOnly')
                ? { showOpenAccessOnly: providedSearchQuery.activeFacets.showOpenAccessOnly === 'true' }
                : /* istanbul ignore next */ {}),
        };
    } else {
        providedSearchQuery.activeFacets = {
            filters: {},
            ranges: {},
        };
    }

    const pageSize = parseInt(providedSearchQuery.pageSize, 10);
    /* istanbul ignore if */
    if (canBulkExport && pageSize === PUB_SEARCH_BULK_EXPORT_SIZE) {
        providedSearchQuery.bulkExportSelected = true;
        providedSearchQuery.pageSize = PUB_SEARCH_BULK_EXPORT_SIZE;
    } else {
        providedSearchQuery.bulkExportSelected = false;
        providedSearchQuery.pageSize = locale.components.sorting.recordsPerPage.indexOf(pageSize) < 0 ? 20 : pageSize;
    }

    providedSearchQuery.sortDirection =
        locale.components.sorting.sortDirection.indexOf(providedSearchQuery.sortDirection) < 0
            ? locale.components.sorting.sortDirection[0]
            : providedSearchQuery.sortDirection;

    providedSearchQuery.sortBy =
        locale.components.sorting.sortBy.map(sortBy => sortBy.value).indexOf(providedSearchQuery.sortBy) < 0
            ? locale.components.sorting.sortBy[1].value
            : providedSearchQuery.sortBy;

    if (!isUnpublishedBufferPage && !!providedSearchQuery.searchQueryParams) {
        delete providedSearchQuery.searchQueryParams.rek_status;
        delete providedSearchQuery.searchQueryParams.rek_created_date;
        delete providedSearchQuery.searchQueryParams.rek_updated_date;
    }

    providedSearchQuery.page =
        (providedSearchQuery.page && /* istanbul ignore next */ parseInt(providedSearchQuery.page, 10)) || 1;

    return providedSearchQuery;
};

export const SearchRecords = ({
    searchQuery,
    publicationsList,
    publicationsListFacets,
    publicationsListPagingData,
    exportPublicationsLoading,
    canUseExport,
    searchLoading,
    searchLoadingError,
    isAdvancedSearch,
    isUnpublishedBufferPage,
    location,
    history,
    actions,
}) => {
    const [state, setState] = React.useState({
        page: 1,
        pageSize: 20,
        sortBy: locale.components.sorting.sortBy[1].value,
        sortDirection: locale.components.sorting.sortDirection[0],
        activeFacets: {
            filters: {},
            ranges: {},
            ...{
                ...(((searchQuery || {}).activeFacets || {}).showOpenAccessOnly === 'true'
                    ? /* istanbul ignore next */ { showOpenAccessOnly: true }
                    : {}),
            },
        },
        advancedSearchFields: [],
        bulkExportSelected: false,
    });

    const updateSemaphore = React.useRef(false);

    const isAdmin = userIsAdmin();
    const isResearcher = userIsResearcher();

    React.useEffect(() => {
        if (!!location && location.search.indexOf('?') >= 0) {
            setState(
                parseSearchQueryStringFromUrl(
                    location.search.substr(1),
                    isResearcher || isAdmin,
                    isUnpublishedBufferPage,
                ),
            );
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [location, isResearcher, isAdmin]);

    React.useEffect(() => {
        // handle browser back button - set state from location/dispatch action for this state
        /* istanbul ignore if */
        if (history.action === 'POP' && /* istanbul ignore next */ location.pathname === pathConfig.records.search) {
            if (location.state) {
                setState(
                    parseSearchQueryStringFromUrl(
                        param(location.state),
                        isResearcher || isAdmin,
                        isUnpublishedBufferPage,
                    ),
                );
            }

            actions.searchEspacePublications(state);
        } else {
            !!location.search &&
                location.search.length > 1 &&
                setState(
                    parseSearchQueryStringFromUrl(
                        location.search.substr(1),
                        isResearcher || isAdmin,
                        isUnpublishedBufferPage,
                    ),
                );
        }

        return actions.clearSearchQuery();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [location, history.action]);

    const updateSearch = () => {
        actions.searchEspacePublications({ ...searchQuery, ...state });
    };

    const updateHistoryAndSearch = () => {
        history.push({
            pathname:
                location.pathname === pathConfig.admin.unpublished
                    ? /* istanbul ignore next */ pathConfig.admin.unpublished
                    : pathConfig.records.search,
            search: param(state),
            state,
        });

        updateSearch();
    };

    React.useEffect(() => {
        updateSemaphore.current && updateHistoryAndSearch();
        updateSemaphore.current = false;
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [state]);

    const pageSizeChanged = pageSize => {
        updateSemaphore.current = true;
        setState({
            ...state,
            pageSize,
            page: 1,
        });
    };

    const pageChanged = page => {
        updateSemaphore.current = true;
        setState({
            ...state,
            page,
        });
    };

    const sortByChanged = (sortBy, sortDirection) => {
        updateSemaphore.current = true;
        setState({
            ...state,
            sortBy,
            sortDirection,
        });
    };

    /* istanbul ignore next */
    const facetsChanged = activeFacets => {
        updateSemaphore.current = true;
        setState({
            ...state,
            activeFacets,
            page: 1,
        });
    };

    const handleExportPublications = exportFormat =>
        actions.exportEspacePublications({
            ...state,
            ...exportFormat,
        });

    /* istanbul ignore next */
    const handleFacetExcludesFromSearchFields = searchFields => {
        const excludesFromLocale = locale.pages.searchRecords.facetsFilter.excludeFacetsList;
        // Iterate the searchfields and add their map from locale into the excluded facets array
        if (searchFields) {
            const importedFacetExcludes = [];
            Object.keys(searchFields).map(key => {
                if (searchFields[key].searchField) {
                    const fieldType =
                        locale.components.searchComponent.advancedSearch.fieldTypes[searchFields[key].searchField];
                    if (fieldType.map) {
                        importedFacetExcludes.push(fieldType.map);
                    }
                }
            });
            setState({
                ...state,
                advancedSearchFields: excludesFromLocale.concat(importedFacetExcludes),
            });
        }
    };

    const txt = locale.pages.searchRecords;
    const pagingData = publicationsListPagingData;
    const isLoadingOrExporting = searchLoading || exportPublicationsLoading;
    const hasSearchParams = !!searchQuery && searchQuery.constructor === Object && Object.keys(searchQuery).length > 0;
    const alertProps = searchLoadingError && {
        ...txt.errorAlert,
        message: txt.errorAlert.message(locale.global.errorMessages.generic),
    };

    return (
        <StandardPage className="page-search-records">
            <Grid container spacing={3}>
                <Grid item xs={12}>
                    <StandardCard className="searchComponent" noHeader standardCardId="search-records-queries">
                        <SearchComponent
                            showAdvancedSearchButton
                            activeFacets={state.activeFacets}
                            searchLoading={searchLoading}
                            clearSearchQuery={actions.clearSearchQuery}
                            updateFacetExcludesFromSearchFields={handleFacetExcludesFromSearchFields}
                            isAdvancedSearch={isAdvancedSearch}
                            isAdmin={isAdmin}
                            isUnpublishedBufferPage={isUnpublishedBufferPage}
                        />
                    </StandardCard>
                </Grid>
                {// first time loading search results
                !hasSearchParams && searchLoading && (
                    <Grid item xs={12}>
                        <InlineLoader message={txt.loadingMessage} loaderId="search-records-loading" />
                    </Grid>
                )}
                {searchLoadingError && (
                    <Grid item xs={12}>
                        <Alert pushToTop {...alertProps} />
                    </Grid>
                )}
                {// no results to display
                hasSearchParams && !searchLoading && publicationsList && publicationsList.length === 0 && (
                    <Grid item xs={12}>
                        <StandardCard {...txt.noResultsFound}>{txt.noResultsFound.text}</StandardCard>
                    </Grid>
                )}
                {// results to display or loading if user is filtering/paging
                (exportPublicationsLoading ||
                    (hasSearchParams && searchLoading) ||
                    (!!publicationsList && publicationsList.length > 0)) && (
                    <Grid item xs sm md={9}>
                        <StandardCard noHeader standardCardId="search-records-results">
                            <Grid container spacing={2} justify="space-between">
                                <Grid item xs="auto">
                                    {pagingData && pagingData.to && pagingData.from && pagingData.total ? (
                                        <span>
                                            {txt.recordCount
                                                .replace('[recordsTotal]', pagingData.total)
                                                .replace('[recordsFrom]', pagingData.from)
                                                .replace('[recordsTo]', pagingData.to)}
                                        </span>
                                    ) : (
                                        <span>{txt.loadingPagingMessage}</span>
                                    )}
                                </Grid>
                                <Grid item xs="auto">
                                    {(isAdmin || isResearcher) && (
                                        <BulkExport
                                            locale={txt.bulkExport}
                                            exportPublications={handleExportPublications}
                                            pageSize={PUB_SEARCH_BULK_EXPORT_SIZE}
                                            totalMatches={publicationsListPagingData.total}
                                        />
                                    )}
                                </Grid>
                                <Grid item xs={12}>
                                    <PublicationsListSorting
                                        sortBy={state.sortBy}
                                        sortDirection={state.sortDirection}
                                        pageSize={state.pageSize}
                                        pagingData={pagingData}
                                        canUseExport={canUseExport}
                                        onSortByChanged={sortByChanged}
                                        onPageSizeChanged={pageSizeChanged}
                                        onExportPublications={handleExportPublications}
                                        disabled={isLoadingOrExporting}
                                    />
                                </Grid>
                                <Grid item xs={12}>
                                    <PublicationsListPaging
                                        loading={isLoadingOrExporting}
                                        pagingData={pagingData}
                                        onPageChanged={pageChanged}
                                        disabled={isLoadingOrExporting}
                                        pagingId="search-records-paging-top"
                                    />
                                </Grid>
                                {isLoadingOrExporting && (
                                    <Grid item xs={12}>
                                        <Grid container justify={'center'}>
                                            <Grid item xs={12}>
                                                <InlineLoader
                                                    message={
                                                        searchLoading
                                                            ? txt.loadingPagingMessage
                                                            : txt.exportPublicationsLoadingMessage
                                                    }
                                                    loaderId="search-records-page-loading"
                                                />
                                            </Grid>
                                        </Grid>
                                    </Grid>
                                )}
                                {!isLoadingOrExporting && publicationsList && publicationsList.length > 0 && (
                                    <Grid item xs={12}>
                                        <RecordsSelectorContext.Provider
                                            value={{
                                                records: publicationsList,
                                            }}
                                        >
                                            <PublicationsList
                                                showAdminActions={isAdmin || isUnpublishedBufferPage}
                                                showUnpublishedBufferFields={isUnpublishedBufferPage}
                                                publicationsList={publicationsList}
                                            />
                                        </RecordsSelectorContext.Provider>
                                    </Grid>
                                )}
                                <Grid item xs={12}>
                                    <PublicationsListPaging
                                        loading={isLoadingOrExporting}
                                        pagingData={pagingData}
                                        onPageChanged={pageChanged}
                                        disabled={isLoadingOrExporting}
                                        pagingId="search-records-paging-bottom"
                                    />
                                </Grid>
                            </Grid>
                        </StandardCard>
                    </Grid>
                )}
                {publicationsListFacets && Object.keys(publicationsListFacets).length !== 0 && (
                    <Hidden smDown>
                        <Grid item md={3}>
                            <StandardRighthandCard title={txt.facetsFilter.title} help={txt.facetsFilter.help}>
                                <FacetsFilter
                                    facetsData={publicationsListFacets}
                                    onFacetsChanged={facetsChanged}
                                    activeFacets={state.activeFacets}
                                    disabled={isLoadingOrExporting}
                                    excludeFacetsList={
                                        (state.advancedSearchFields &&
                                            state.advancedSearchFields.length &&
                                            /* istanbul ignore next */
                                            state.advancedSearchFields) ||
                                        locale.pages.searchRecords.facetsFilter.excludeFacetsList
                                    }
                                    renameFacetsList={txt.facetsFilter.renameFacetsList}
                                    lookupFacetsList={txt.facetsFilter.lookupFacetsList}
                                    showOpenAccessFilter
                                />
                            </StandardRighthandCard>
                        </Grid>
                    </Hidden>
                )}
            </Grid>
        </StandardPage>
    );
};

SearchRecords.propTypes = {
    searchQuery: PropTypes.object,
    publicationsList: PropTypes.array,
    publicationsListFacets: PropTypes.object,
    publicationsListPagingData: PropTypes.object,
    exportPublicationsLoading: PropTypes.bool,
    canUseExport: PropTypes.bool,
    searchLoading: PropTypes.bool,
    searchLoadingError: PropTypes.bool,
    isAdvancedSearch: PropTypes.bool,
    isUnpublishedBufferPage: PropTypes.bool,

    location: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired,
    actions: PropTypes.object,
};

export default SearchRecords;
