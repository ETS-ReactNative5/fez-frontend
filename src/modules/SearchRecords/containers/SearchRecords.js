import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchRecords from '../components/SearchRecords';
import * as actions from 'actions';
import { withRouter } from 'react-router-dom';
import { routes } from 'config';

const mapStateToProps = (state, props) => ({
    ...state.get('searchRecordsReducer'),
    ...state.get('exportPublicationsReducer'),
    canUseExport: true,
    isUnpublishedBufferPage: props.location.pathname === routes.pathConfig.admin.unpublished,
    isAdmin: !!state && !!state.get('accountReducer') && !!state.get('accountReducer').account && state.get('accountReducer').account.canMasquerade,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators(actions, dispatch),
    };
}

let SearchRecordsContainer = connect(mapStateToProps, mapDispatchToProps)(SearchRecords);
SearchRecordsContainer = withRouter(SearchRecordsContainer);

export default SearchRecordsContainer;
