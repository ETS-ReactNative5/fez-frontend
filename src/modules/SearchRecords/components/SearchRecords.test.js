import SearchRecords from './SearchRecords';

function setup(testProps, isShallow = true) {
    const props = {
        ...testProps,
        exportPublicationsLoading: false,
    };
    return getElement(SearchRecords, props, isShallow);
}

describe('SearchRecords page', () => {

    it('should render placeholders', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders loading screen while export publications loading', () => {
        const wrapper = setup({ exportPublicationsLoading: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
