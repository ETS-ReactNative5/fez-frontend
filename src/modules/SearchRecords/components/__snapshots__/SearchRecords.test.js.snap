// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchRecords page renders loading screen while export publications loading 1`] = `
<StandardPage
  className="page-search-records"
  title="eSpace search"
>
  <StandardCard
    className="search-component"
  >
    <withRouter(Connect(SearchComponent))
      className="search-body"
    />
  </StandardCard>
  <div
    className="columns"
  />
</StandardPage>
`;

exports[`SearchRecords page should render placeholders 1`] = `
<StandardPage
  className="page-search-records"
  title="eSpace search"
>
  <StandardCard
    className="search-component"
  >
    <withRouter(Connect(SearchComponent))
      className="search-body"
    />
  </StandardCard>
  <div
    className="columns"
  />
</StandardPage>
`;
